hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1060,6 +1074,44 @@ public void updateMembers(Collection<MemberInfo> members) {
         }
     }
 
+    private boolean shouldProcessMemberUpdate(Map<Address, MemberImpl> currentMembers,
+                                              Collection<MemberInfo> newMemberInfos) {
+
+        int currentMembersSize = currentMembers.size();
+        int newMembersSize = newMemberInfos.size();
+
+        if (currentMembersSize > newMembersSize) {
```

Comment:
as discussed - there is still the problem when 2 members disappear and 1 new member joins.